app/models/job.ts

```diff
@@ -226,6 +226,40 @@ export const statesToDefaultMessages: any = Object.values(stateMachine.states).r
 
 const defaultMessages = Object.values(statesToDefaultMessages);
 
+
+/**
+ * When the job status is updated we might want to remove some of the prior message parts
+ * while retaining some of the information from the message. For example, we might want
+ * "The job is generating a preview before auto-pausing. The CMR identified 10 granules." to become
+ * "The CMR identified 10 granules." when a job goes from previewing to running.
+ * @param message - the job message that may need changing
+ * @param partsToRemove - an array of strings to remove (sentences without periods--e.g. ["the job is paused"])
+ * @param partsFilter - an optional filter for removal of message parts that do not match this filter
+ * @returns the new message, with partsToRemove removed and sentence structure maintained,
+ * or an empty string if the new message has no retained parts
+ */
+function removeMessageParts(
+  message: string,
+  partsToRemove,
+  partsFilter: (p: string) => boolean = undefined,
+): string {
+  if (!message) {
+    return;
```

Comment:
I think this `return` might break your contract of returning an empty string if there are no retained parts (I think it returns null or undefined).